content/blog/2021-04-19-experiment-refs.md

```diff
@@ -0,0 +1,277 @@
+[two sets of commands](https://git-scm.com/book/en/v2/Git-Internals-Plumbing-and-Porcelain):
+the commonly used user-friendly “porcelain” commands (like `git checkout`) and
+the lower level “plumbing” commands._
+
+This naturally begs the question, “What is `exp-9b1a5`?”
+
+The answer is simple, it’s a custom DVC Git ref pointing to a Git commit:
+
+```dvc
+$ git show-ref exp-9b1a5
+7365395a722ab30ddc8ff28536b5e50371006348 refs/exps/04/13348986d70a3b1fb0061b7217866b60a89390/exp-9b1a5
+```
+
+_Note: that `dvc exp show --sha` can be used to view Git commit SHAs for
+experiments. Using DVC experiments should never require you to use any of the
+low-level Git plumbing commands like `git show-ref`._
+
+If we examine the experiment commit itself, we can see that it is just a regular
+commit object that contains our hyperparameter change and the results of the
+run:
+
+```dvc
+$ git show 7365395
+commit 7365395a722ab30ddc8ff28536b5e50371006348 (refs/exps/04/13348986d70a3b1fb0061b7217866b60a89390/exp-9b1a5)
+Author: Peter Rowlands <peter@pmrowla.com>
+Date:   Tue Apr 6 07:20:17 2021 +0000
+
+    dvc: commit experiment 9b1a52d692cf9a725ee790ae61a376fa470e8589ec1cf90385a459a92b3a0544
+
+diff --git a/dvc.lock b/dvc.lock
+index a55afd2..4210d68 100644
+--- a/dvc.lock
++++ b/dvc.lock
+@@ -30,19 +30,19 @@ stages:
+       size: 2480
+     params:
+       params.yaml:
+-        featurize.max_features: 3000
++        featurize.max_features: 2000
+         featurize.ngrams: 2
+...
+diff --git a/scores.json b/scores.json
+index 27f6dab..8270914 100644
+--- a/scores.json
++++ b/scores.json
+@@ -1,4 +1,4 @@
+ {
+-    "avg_prec": 0.6040544652105823,
+-    "roc_auc": 0.9608017142900953
++    "avg_prec": 0.5858888885424922,
++    "roc_auc": 0.944996664954421
+ }
+...
+```
+
+## DVC and custom Git refs
+
+In DVC 2.0, we now use the custom `refs/exps` namespace for storing DVC
+experiments in Git. Under the hood, using Git refs allows us to keep using all
+of the versioning capabilities provided by Git, without polluting your
+repository with actual Git branches and tags. Since the user-friendly Git
+porcelain commands (like `git checkout` and `git diff`) only resolve branches
+and tags (and will ignore custom references), DVC experiments are essentially
+hidden from your Git repository (and only visible to DVC commands).
+
+Even though the experiment refs themselves are “invisible” to Git porcelain
+commands, Git commit SHAs for experiments can be used in any Git command. This
+allows you to leverage the power of tools like `git diff` to compare things like
+code changes between a DVC experiment and any other Git commit (meaning you can
+even compare experiment commit SHAs to Git branches or tags).
+
+Likewise, for tools which provide a GUI on top of Git, experiments will be
+hidden from your repository in typical use cases:
+
+![`gitk --branches --tags` example](/uploads/images/2021-04-19/gitk-branches-tags.png 'gitk --branches --tags')
```

Comment:
- images are too small (on retina screens?) - can we make a high res large screenshots?
- to make `gitk --branches --tags` a subtitle - put after the `) _gitk --branches --tags_` ?

Reply:
💯 
I see nothing in the original one 🤓  and it is hard to read even if I click to it.